pages/security/output_encoding.md

```diff
@@ -0,0 +1,43 @@
++---
++title: Output Enco
++permalink: /security/output-enciding
++layout: docs
++sidenav: security
++---
+
+# Output encoding
+
+## What is cross site scripting (XSS)?
+
+Cross site scripting or XSS is a form of attack on a web application which involves executing code on a user's browser. Output encoding is a defense against XSS attacks.
+
+To get a more extensive understanding of XSS, see [excess xss](https://excess-xss.com/).
+
+## How to correctly encode output
+
+Protecting from XSS attacks requires developers to consider how data is being displayed on a page. If the data could come from a user's input in any way (including through the site's URL), then correct encoding of the output has to be considered.
+
+Since most web applications at 18F are built through JavaScript or backend frameworks, this guide will go over output encoding issues by those frameworks in addition to plain JavaScript.
+
+### Plain JavaScript
+
+When writing plain JavaScript, developer's have to consider where data is coming from whenever it's being output in the web application. While data that comes from a backend database is usually considered as needing output encoding, code also could need output encoding when extracting data from the current page's url (which an attacker could modify and send to a user).
+
+The [excess xss prevention](https://excess-xss.com/#xss-prevention) section has up-to-date information on how to prevent XSS attacks in all the possible ways they could happen.
+
+### React
+
+By default React DOM esacpes all output. This means that output in JSX components will usually be safe, which is great news. This is discussed in [React's documentation](https://reactjs.org/docs/introducing-jsx.html#jsx-prevents-injection-attacks).
+
+There are some cases where ouput may not be correctly escaped in React components. Here are some of those cases:
+
+- Using the `dangerouslySetInnerHTML` prop. It's named this for a reason
```

Comment:
Add period after reason

Reply:
I just read in our [content guide that lists should not have punctuation](https://content-guide.18f.gov/punctuation/#bulleted-lists).

I removed periods from the other list items.